src/plugins/intel_gpu/src/plugin/ops/softmax.cpp

```diff
@@ -14,34 +14,33 @@
 namespace ov {
 namespace intel_gpu {
 
-static cldnn::softmax::dimension_t GetSoftmaxAxis(int64_t axis, size_t rank) {
-    switch (axis) {
-    case 0: return cldnn::softmax::normalize_b;
-    case 1: return cldnn::softmax::normalize_f;
-    case 2:
-        if (rank > 4)
-            return cldnn::softmax::normalize_z;
-        else
-            return cldnn::softmax::normalize_y;
-    case 3:
-        if (rank > 4)
-            return cldnn::softmax::normalize_y;
-        else
-            return cldnn::softmax::normalize_x;
-    case 4:
-        return cldnn::softmax::normalize_x;
-    default: IE_THROW() << "Invalid softmax axis " << axis;
-    }
-    return cldnn::softmax::normalize_fyx;
+static void CreateSoftmaxOp(Program& p, const std::shared_ptr<ngraph::op::v1::Softmax>& op) {
+    p.ValidateInputs(op, {1});
+    auto inputPrimitives = p.GetInputPrimitiveIDs(op);
+    std::string layerName = layer_type_name_ID(op);
+    auto softmaxPrim = cldnn::softmax(layerName,
+                                      inputPrimitives[0],
+                                      op->get_axis(),
+                                      op->get_friendly_name());
+    p.AddPrimitive(softmaxPrim);
+    p.AddPrimitiveToProfiler(op);
 }
 
-static void CreateSoftmaxOp(Program& p, const std::shared_ptr<ngraph::op::v1::Softmax>& op) {
+static void CreateSoftmaxOp(Program& p, const std::shared_ptr<ngraph::op::v8::Softmax>& op) {
     p.ValidateInputs(op, {1});
     auto inputPrimitives = p.GetInputPrimitiveIDs(op);
     std::string layerName = layer_type_name_ID(op);
+
+    int64_t axis = op->get_axis();
+    size_t rank = op->get_input_shape(0).size();
+    if (axis < 0)
+        axis += rank;
+    if (axis < 0 || axis >= static_cast<int64_t>(rank))
```

Comment:
Why it's needed? The axis is supposed to be normalized later in the plugin

Reply:
The axis is normalized in the same place as in other operations (for example: gather_elements, cum_sum, interpolate)